test/e2e/config/vsphere-ci.yaml

```diff
@@ -28,10 +28,12 @@ providers:
   - name: cluster-api
     type: CoreProvider
     versions:
-      - name: v0.3.11
+      - name: v0.4.0
         # Use manifest from source files
-        value: "https://github.com/kubernetes-sigs/cluster-api/releases/download/v0.3.11/core-components.yaml"
+        value: "https://storage.googleapis.com/artifacts.k8s-staging-cluster-api.appspot.com/components/nightly_master_20210310/core-components.yaml"
```

Comment:
this means we're moving our e2e to test against CAPI main-nightly, let's keep an eye on e2e if something breaks

Reply:
Technically we are not moving to the latest nightly-build, and it is fixed to a particular SHA. I ran the E2E tests with this image, and they work. We might have to keep moving this regularly to the latest available nightly build.